pontoon/base/views.py

```diff
@@ -988,50 +994,119 @@ def transvision(request):
         return HttpResponse("error")
 
 
+def download_file(prefixes, dirnames, relative_path):
+    for prefix in prefixes:
+        for dirname in dirnames:
+            url = os.path.join(prefix.replace('{locale_code}', dirname), relative_path)
+            r = requests.get(url, stream=True)
+            if not r.ok:
+                continue
+
+            filename = str(uuid.uuid4()) + os.path.splitext(url)[1]
```

Comment:
The reason I didn't use tempfile is because it's gone after you close it (at the end of with clause).

I'll see how risky modifying the parsers to accept content would be.